### ISSUE TITLE
can't open file run_ros2_batch.py

### ISSUE THREAD

**Issue by dirk-thomas** [ros2/build_farmer#177]
Some aarch64 jobs:

* https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_debug/783/
* https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_debug/779/
* https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_repeated/747/
* https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_repeated/745/
* https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_repeated/744/

failed with:

> python3: can't open file 'run_ros2_batch.py': [Errno 2] No such file or directory

The workspace does contain the file `run_ros2_batch.py` though.

None of the amd64 Linux jobs shows the same problem.

---

Reported during #168.

**Comment by dirk-thomas** [comment]
@nuclearsandwich Will try to update the arm nodes with current versions of system packages (especially `docker`).

**Comment by nuclearsandwich** [comment]
> @nuclearsandwich Will try to update the arm nodes with current versions of system packages

The host has been upgraded and the current Docker version is 18.09.

**Comment by dirk-thomas** [comment]
Unfortunately one of the nightly builds still failed with the same reason: https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_repeated/752/

**Comment by nuclearsandwich** [comment]
I've deployed new linux-arm hosts to the CI farm which can run both armhf and arm64(aka aarch64) jobs. I've configured them to run both to see if the new hosts also hit this problem. If we have problems with the armhf jobs getting starved out because Jenkins is running aarch64 jobs on them when it could use the dedicated aarch64 host I'll pull the aarch64 label off of the deployment group.

**Comment by clalancette** [comment]
I haven't seen this issue on aarch64 debug or repeated since the beginning of May.  Given that, I'm going to close this out, assuming it is fixed by the new hosts.  We can reopen if it comes back.

**Comment by nuclearsandwich** [comment]
This failure has occurred a number of times in the last month

* https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/876/
* https://ci.ros2.org/job/nightly_linux-aarch64_repeated/864/
* https://ci.ros2.org/job/nightly_linux-aarch64_debug/880/
* https://ci.ros2.org/job/nightly_linux-aarch64_extra_rmw_release/412/

I haven't seen the issue once on the on the AWS a1.xlarge instances. I'd also like to try one of the Packet.net c2.arm.large instances now that those appear to be available.

Even if I knew what to trace to investigate this more closely I'm not sure it'd be worth the effort versus retiring this host in favor of the other options.

**Comment by nuclearsandwich** [comment]
The new Packet.net machine has exhibited the same failure running armhf jobs https://ci.ros2.org/job/nightly_linux-armhf_debug/198/console#console-section-6
https://ci.ros2.org/job/nightly_linux-armhf_release/202/console#console-section-6

**Comment by emersonknapp** [comment]
@nuclearsandwich do we have any leads on what's causing this?

**Comment by nuclearsandwich** [comment]
> @nuclearsandwich do we have any leads on what's causing this?

No. I can't reproduce it reliably enough to force the case and trace it. I did see something on my local workstation that was very similar on the surface level earlier this week. Running the a ros_buildfarm `generate_ci_script.py` script I got an issue where a script wasn't runnable within the docker container, but that turned out to be because the remote that ros_buildfarm was being cloned from during the job was out-of-sync with the local repository the script was generated from. But I don't think there's an actual analogue there with what's happening on this packet host.

One thing that _just_ occurred to me is that the packet.net host because of its size is the only one we run multiple executors on. That gives me ideas for both possible temporary mitigation (reduce to 1 executor) and how to try and trigger a reproduction to investigate.
